torch/_decomp/decompositions.py

```diff
@@ -4343,6 +4343,14 @@ def _weight_norm_interface(x, y, dim):
     norm = x.norm(2, keep_dim, keepdim=True)
     return x * (y / norm), norm
 
+@register_decomposition(aten.put)
+def put(self, index, source, accumulate=False):
+    flattened = self.flatten().clone()
+    if accumulate:
+        flattened[index] += source
+    else:
+        flattened[index] = source
```

Comment:
Mutation isn't legal inside decompositions of functional operators. If I'm not mistaken you could just use `torch.index_put` though.
```suggestion
    flattened = self.flatten()
    flattened = torch.index_put(flattened, [index], source, accumulate)
```

Reply:
It seems this fails because `put` supports NumPy-style "reverse broadcasting":

```py
>>> x = torch.tensor(1.0)
>>> index = torch.tensor(0)
>>> source = torch.tensor([0.0])
>>> torch.put(x, index, source)
tensor(0.)
```

Reply:
I guess `source.reshape(index.shape)` should work, assuming the actual shape conditions are all tested separately.